### DISCUSSION TITLE
OP - FX Price Acceptance Period

### DISCUSSION THREAD

**Discussion by martuOW** [OpenWealth/OpenWealth#166] [Q&A] [answered]
The new endpoint to request a FX rate for a specific amount triggers several possible workflows:

- FX rate is accepted within validity period
- FX rate is accepted after validity
- FX rate is not accepted

For each situation we need to define a standard behaviour between the parties. Since unlike in the case of a web-base application we do not have full control of the validity period. There is a risk that even when the FX rate was accapted within the period, by the time the response arrives at the SP, the period has elapsed. These scenarios (and more) have to be identified and we need to come to a common understanding on how to handle the timing over the API.

--> Labels: Order Placement, question

**Comment by dustyush0sq** [comment, marked as answer]
To my understanding and according to previous discussions, this could this could be potentially addressed with two parameters of returned quote:   quote type and expiry date

Expiry date would indicate timestamp until which the quote would be valid for trading. Past the expiry date the sell side would not guarantee execution price

Quote type - indicative or tradeable  - would indicate whether the sell side  guarantees execution price. 

Therefore, if an order previously quoted is submitted against tradeable quote within expiry time interval, then the sell side is supposed to guarantee execution price, otherwise no guarantees are supposed to given in general case
